diff --git a/includes/options/class-amp-options-manager.php b/includes/options/class-amp-options-manager.php
index d590b9a63..b314bb5fe 100644
--- a/includes/options/class-amp-options-manager.php
+++ b/includes/options/class-amp-options-manager.php
@@ -33,7 +33,7 @@ class AMP_Options_Manager {
 		Option::ANALYTICS               => [],
 		Option::ALL_TEMPLATES_SUPPORTED => true,
 		Option::SUPPORTED_TEMPLATES     => [ 'is_singular' ],
-		Option::VERSION                 => AMP__VERSION,
+		Option::VERSION                 => WEBSTORIES_AMP_VERSION,
 		Option::READER_THEME            => ReaderThemes::DEFAULT_READER_THEME,
 		Option::PLUGIN_CONFIGURED       => false,
 	];
@@ -403,7 +403,7 @@ class AMP_Options_Manager {
 		$options = apply_filters( 'amp_options_updating', $options, $new_options );
 
 		// Store the current version with the options so we know the format.
-		$options[ Option::VERSION ] = AMP__VERSION;
+		$options[ Option::VERSION ] = WEBSTORIES_AMP_VERSION;
 
 		return $options;
 	}
