diff --git a/node_modules/draft-js-import-element/esm/stateFromElement.js b/node_modules/draft-js-import-element/esm/stateFromElement.js
index eccabfd..9ec7496 100644
--- a/node_modules/draft-js-import-element/esm/stateFromElement.js
+++ b/node_modules/draft-js-import-element/esm/stateFromElement.js
@@ -373,7 +373,7 @@ function () {
         switch (customInline.type) {
           case 'STYLE':
             {
-              style = style.add(customInline.style);
+             [].concat(customInline.style).forEach(customStyle => { style = style.add(customStyle); });
               break;
             }
 
diff --git a/node_modules/draft-js-import-element/lib/stateFromElement.js b/node_modules/draft-js-import-element/lib/stateFromElement.js
index e814c05..c469bda 100644
--- a/node_modules/draft-js-import-element/lib/stateFromElement.js
+++ b/node_modules/draft-js-import-element/lib/stateFromElement.js
@@ -389,7 +389,7 @@ function () {
         switch (customInline.type) {
           case 'STYLE':
             {
-              style = style.add(customInline.style);
+              [].concat(customInline.style).forEach(customStyle => { style = style.add(customStyle); });
               break;
             }
 
