diff --git a/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js b/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js
index 6ef5af8..2f11acb 100644
--- a/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js
+++ b/node_modules/eslint-plugin-react-hooks/cjs/eslint-plugin-react-hooks.development.js
@@ -739,6 +739,9 @@ var ExhaustiveDeps = {
         additionalHooks: {
           type: 'string'
         },
+        additionalHooksMap: {
+          type: 'object'
+        },
         enableDangerousAutofixThisMayCauseInfiniteLoops: {
           type: 'boolean'
         }
@@ -748,9 +751,11 @@ var ExhaustiveDeps = {
   create: function (context) {
     // Parse the `additionalHooks` regex.
     var additionalHooks = context.options && context.options[0] && context.options[0].additionalHooks ? new RegExp(context.options[0].additionalHooks) : undefined;
+    var additionalHooksMap = context.options && context.options[0] && context.options[0].additionalHooksMap || undefined;
     var enableDangerousAutofixThisMayCauseInfiniteLoops = context.options && context.options[0] && context.options[0].enableDangerousAutofixThisMayCauseInfiniteLoops || false;
     var options = {
       additionalHooks: additionalHooks,
+      additionalHooksMap: additionalHooksMap,
       enableDangerousAutofixThisMayCauseInfiniteLoops: enableDangerousAutofixThisMayCauseInfiniteLoops
     };
 
@@ -2181,7 +2186,9 @@ function getReactiveHookCallbackIndex(calleeNode, options) {
       return 1;
 
     default:
-      if (node === calleeNode && node.name.match(/^use.+Effect$/)) {
+      if (node === calleeNode && options && options.additionalHooksMap && node.name in options.additionalHooksMap) {
+        return options.additionalHooksMap[node.name];
+      } else if (node === calleeNode && node.name.match(/^use.+Effect$/)) {
         return 0;
       } else if (node === calleeNode && options && options.additionalHooks) {
         // Allow the user to provide a regular expression which enables the lint to
