diff --git a/node_modules/expect-puppeteer/dist/index.js b/node_modules/expect-puppeteer/dist/index.js
index 9db58ab..dd27744 100644
--- a/node_modules/expect-puppeteer/dist/index.js
+++ b/node_modules/expect-puppeteer/dist/index.js
@@ -4,13 +4,13 @@ const checkIsPuppeteerInstance = (instance)=>{
     return Boolean(instance?.constructor?.name && typeof instance === "object" && "$" in instance);
 };
 const checkIsPage = (instance)=>{
-    return checkIsPuppeteerInstance(instance) && instance?.constructor?.name === "CDPPage";
+    return checkIsPuppeteerInstance(instance) && (instance?.constructor?.name === "CDPPage" || instance?.constructor?.name === "CdpPage");
 };
 const checkIsFrame = (instance)=>{
-    return checkIsPuppeteerInstance(instance) && instance?.constructor?.name === "Frame";
+    return checkIsPuppeteerInstance(instance) && (instance?.constructor?.name === "CDPFrame" || instance?.constructor?.name === "CdpFrame");
 };
 const checkIsElementHandle = (instance)=>{
-    return checkIsPuppeteerInstance(instance) && instance?.constructor?.name === "CDPElementHandle";
+    return checkIsPuppeteerInstance(instance) && (instance?.constructor?.name === "CDPElementHandle" || instance?.constructor?.name === "CdpElementHandle");
 };
 const getContext = async (instance, pageFunction)=>{
     if (checkIsFrame(instance) || checkIsPage(instance)) {
