diff --git src/PhpParser/NodeVisitor/UseStmt/UseStmtCollection.php src/PhpParser/NodeVisitor/UseStmt/UseStmtCollection.php
index ec64786..74e40d3 100644
--- src/PhpParser/NodeVisitor/UseStmt/UseStmtCollection.php
+++ src/PhpParser/NodeVisitor/UseStmt/UseStmtCollection.php
@@ -53,7 +53,7 @@ final class UseStmtCollection implements IteratorAggregate
      * @var array<string, list<Use_>>
      */
     private array $nodes = [
-        null => [],
+        '' => [],
     ];
 
     public function add(?Name $namespaceName, Use_ $use): void
