Index: src/Generator.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Generator.php b/src/Generator.php
--- a/src/Generator.php	(revision 0dfddc236629eb7ead4df2dfa3d79d548e011a3b)
+++ b/src/Generator.php	(date 1626263688085)
@@ -959,7 +959,7 @@
         }
 
         if ($method->hasReturnType()) {
-            $returnType = (string) $method->getReturnType();
+            $returnType = PHP_VERSION_ID >= 70100 ? $method->getReturnType()->getName() : (string) $method->getReturnType();
         } else {
             $returnType = '';
         }
@@ -1149,7 +1149,7 @@
                     $nullable = '?';
                 }
 
-                if ($parameter->hasType() && (string) $parameter->getType() !== 'self') {
+                if (PHP_VERSION_ID <= 70100 && $parameter->hasType() && (string) $parameter->getType() !== 'self') {
                     $typeDeclaration = (string) $parameter->getType() . ' ';
                 } elseif ($parameter->isArray()) {
                     $typeDeclaration = 'array ';
