diff --git a/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js b/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js
index d819547..b833ddb 100644
--- a/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js
+++ b/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js
@@ -213,8 +213,10 @@ class ExecutionContext {
             throw error;
         }
         const { exceptionDetails, result: remoteObject, } = await callFunctionOnPromise.catch(rewriteError);
-        if (exceptionDetails)
+        if (exceptionDetails) {
+          console.log(functionText, exceptionDetails);
             throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));
+        }
         return returnByValue
             ? helper_js_1.helper.valueFromRemoteObject(remoteObject)
             : JSHandle_js_1.createJSHandle(this, remoteObject);
