diff --git a/node_modules/react-photo-gallery/dist/react-photo-gallery.esm.js b/node_modules/react-photo-gallery/dist/react-photo-gallery.esm.js
index 4421732..775f4b2 100644
--- a/node_modules/react-photo-gallery/dist/react-photo-gallery.esm.js
+++ b/node_modules/react-photo-gallery/dist/react-photo-gallery.esm.js
@@ -534,7 +534,7 @@ var Gallery = React.memo(function Gallery(_ref) {
       // only do something if width changes
       var newWidth = entries[0].contentRect.width;
 
-      if (containerWidth !== newWidth) {
+      if (newWidth !== 0 && containerWidth !== newWidth) {
         // put in an animation frame to stop "benign errors" from
         // ResizObserver https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded
         animationFrameID = window.requestAnimationFrame(function () {
diff --git a/node_modules/react-photo-gallery/dist/react-photo-gallery.umd.js b/node_modules/react-photo-gallery/dist/react-photo-gallery.umd.js
index 45cec13..52a1552 100644
--- a/node_modules/react-photo-gallery/dist/react-photo-gallery.umd.js
+++ b/node_modules/react-photo-gallery/dist/react-photo-gallery.umd.js
@@ -540,7 +540,7 @@
         // only do something if width changes
         var newWidth = entries[0].contentRect.width;
 
-        if (containerWidth !== newWidth) {
+        if (newWidth !== 0 && containerWidth !== newWidth) {
           // put in an animation frame to stop "benign errors" from
           // ResizObserver https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded
           animationFrameID = window.requestAnimationFrame(function () {
